data/components/frame-list.js

```diff
@@ -182,9 +183,25 @@ var FrameBubble = React.createFactory(React.createClass({
     }
 
     if (!preview && frame.mqtt) {
+      var mqtt = frame.mqtt;
+      var data;
+      op = "MQTT " + mqtt.cmd;
+      if (mqtt.cmd === "publish") {
+        payload = mqtt.topic;
+        data = mqtt.payload;
+      } else {
+        payload = mqtt.messageId || mqtt.clientId || mqtt.returnCode;
+        data = {packet: mqtt};
+        if (mqtt.subscriptions)
+          data = {subscriptions: mqtt.subscriptions};
+        if (mqtt.granted)
+          data = {granted: mqtt.granted};
+        if (mqtt.unsubscriptions)
+          data = {unsubscriptions: mqtt.unsubscriptions};
+      }
       preview = TreeView({
         key: "preview-mqtt",
-        data: {"MQTT": frame.mqtt},
+        data: data,
```

Comment:
Well, to be blunt, it is important to be consistent :)

You could have several connections with different protocols on the same page. For example, in our development environment we use JSON messages, but the development server uses SocketIO to do live reloading - and they both show up in the panel at the same time. It would be strange if they had different view paradigms.

Ultimately, I think a review of the efficiency of the user interface is welcome. Extraneous clicks are rarely useful. I vote that you do what you suggested and move it into another branch. Then you can send a pull request for it separately so that honza can look at it; it is ultimately his decision. For this PR, let's keep it simple and just add the MQTT protocol support, keeping the protocols consistent. Agree?